src/components/plots/barPlot.ts

```diff
@@ -151,6 +166,114 @@ export module Plot {
+     * of all bars, we return the closest in the dominant direction (x for horizontal
+     * charts, y for vertical) and break ties
+     *
+     * @param {Point} queryPoint The point to which dataset points should be compared
+     *
+     * @returns {PlotData} The PlotData closest to queryPoint
+     */
+    public getClosestPlotData(queryPoint: Point): PlotData {
+      var keys = this.datasetOrder();
+      var chartXExtent = { min: 0, max: this.width() };
+      var chartYExtent = { min: 0, max: this.height() };
+
+      var minDist = Infinity;
+      var closest: any;
+
+      var accessor: { [axis: string]: (p: Point) => number } = {
+        "x": (p) => p.x,
+        "y": (p) => p.y
+      };
+
+      var dominantAxis: string = this._isVertical ? "x" : "y";
+      var secondaryAxis: string = this._isVertical ? "y" : "x";
+
+      keys.forEach((key) => {
+        var plotData = this.getAllPlotData(key);
+        if (plotData != null && plotData.pixelPoints != null) {
+          plotData.pixelPoints.forEach((plotPt, i) => {
+            var bar = plotData.selection[0][i];
+
+            if (!Bar._intersectsBBox(chartXExtent, chartYExtent, bar.getBBox())) {
+              // bar isn't visible on plot; ignore it
+              return;
+            }
+
+            var dist = Infinity;
+
+            if (Bar._intersectsBBox(queryPoint.x, queryPoint.y, bar.getBBox())) {
+              // queryPoint is inside of this bar
+              dist = -Infinity;
+            } else {
+              dist = Math.abs(accessor[dominantAxis](queryPoint) - accessor[dominantAxis](plotPt));
+            }
+
+            // if we find a closer bar, record its distance and start a new candidate list
+            if (dist < minDist) {
+              closest = [];
+              minDist = dist;
+            }
+
+            // bars minDist away are part of the closest set
+            if (dist === minDist) {
+              closest.push({
+                datum: plotData.data[i],
+                pixelPoint: plotPt,
+                node: bar
+              });
+            }
+          });
+        }
+      });
+
+      if (closest == null || closest.length === 0) {
+        return {data: [], pixelPoints: [], selection: d3.select()};
+      }
+
+      if (minDist >= 0) {
+        // minDist >= 0 implies we're not inside of any bars
+        // so, pick the one we're closest to in the secondary direction
+        var secondaryClosest: any;
+        minDist = Infinity;
+
+        closest.forEach((candidate: any, i: number) => {
+          var dist = Math.abs(accessor[secondaryAxis](queryPoint) - accessor[secondaryAxis](candidate.pixelPoint));
+
+          if (dist < minDist) {
+            secondaryClosest = [];
+            minDist = dist;
+          }
+
+          if (dist === minDist) {
+            secondaryClosest.push(candidate);
+          }
+        });
+
+        closest = secondaryClosest;
+      }
```

Comment:
The above logic looks extremely similar to the logic dealing with being inside bars.  Can these two pieces of logic be collapsed?